src/models/pnid.ts

```diff
@@ -223,18 +229,53 @@ PNIDSchema.method('scrub', async function scrub() {
 	// * Remove all personal info from a PNID
 	// * Username and PID remain so thye do not get assigned again
 
-	if (this.connections?.stripe?.subscription_id) {
+	if (this.connections?.stripe?.customer_id) {
 		try {
+			const customerID = this.connections.stripe.customer_id;
+
 			if (stripe) {
-				await stripe.subscriptions.del(this.connections.stripe.subscription_id);
+				const customer = await stripe.customers.retrieve(customerID);
+
+				if (!customer.deleted) {
+					// * Deleting will also cancel subscriptions automatically
+					await stripe.customers.del(customerID);
+				}
 			} else {
-				LOG_WARN(`SCRUBBING USER DATA FOR USER ${this.username}. HAS STRIPE SUBSCRIPTION ${this.connections.stripe.subscription_id}, BUT STRIPE CLIENT NOT ENABLED. SUBSCRIPTION NOT CANCELED`);
+				LOG_WARN(`SCRUBBING USER DATA FOR USER ${this.username}. HAS STRIPE DATA UDER ID ${this.connections.stripe.customer_id}, BUT STRIPE CLIENT NOT ENABLED.`);
 			}
 		} catch (error) {
-			LOG_ERROR(`ERROR REMOVING ${this.username} STRIPE SUBSCRIPTION. ${error}`);
+			LOG_ERROR(`ERROR REMOVING ${this.username} STRIPE DATA. ${error}`);
+		}
+	}
+
+	if (config.discourse.forum_url && config.discourse.api_key && config.discourse.api_username) {

```

Comment:
There's error handling for miiverse (log and continue) but not for forum. Is that intentional?

Reply:
Since it's just using fetch I didn't think there was a path where this would throw anything, is that not the case? Stripe will throw on certain errors rather than just returning which is why there's error handling there

Reply:
Fetch can throw errors on network issues or if the body is not json parsable (like if cloudflare shows a gateway timeout error or smth)

Reply:
Oh right I didn't even think about that